sound/soc/sof/intel/hda.c

```diff
@@ -1207,21 +1209,24 @@ static bool link_slaves_found(struct snd_sof_dev *sdev,
 
 			if (ids[j].link_id != link_id ||
 			    ids[j].id.part_id != part_id ||
-			    ids[j].id.mfg_id != mfg_id)
+			    ids[j].id.mfg_id != mfg_id ||
+			    ids[j].id.sdw_version != version)
 				continue;
 
 			/* find out how many identical parts are expected */
 			for (k = 0; k < link->num_adr; k++) {
 				u64 adr2 = link->adr_d[k].adr;
-				unsigned int part_id2, link_id2, mfg_id2;
+				unsigned int part_id2, link_id2, mfg_id2, version2;
 
 				mfg_id2 = SDW_MFG_ID(adr2);
 				part_id2 = SDW_PART_ID(adr2);
 				link_id2 = SDW_DISCO_LINK_ID(adr2);
+				version2 = SDW_VERSION(adr2);
 
 				if (link_id2 == link_id &&
 				    part_id2 == part_id &&
-				    mfg_id2 == mfg_id)
+				    mfg_id2 == mfg_id &&
+				    version2 == version)
 					expected_part_count++;
```

Comment:
if all the field of adr are compared how about to use  if ( adr == adr2) ? maybe some fileds are not used, we can use a mask to remove them

Reply:
> if all the field of adr are compared how about to use if ( adr == adr2) ? maybe some fileds are not used, we can use a mask to remove them

We can't compare the unique_id field here. And if we want to use mask, the code will not look simpler or say shorter. So I prefer to leave it as it is.

Reply:
why ? we don't need following,   just if (MASK_WITOUTH_UNIQUE_ID(adr) == MASK_WITOUTH_UNIQUE_ID(adr2))
````
				mfg_id2 = SDW_MFG_ID(adr2);
				part_id2 = SDW_PART_ID(adr2);
				link_id2 = SDW_DISCO_LINK_ID(adr2);
				version2 = SDW_VERSION(adr2);

				if (link_id2 == link_id &&
				    part_id2 == part_id &&
				    mfg_id2 == mfg_id)
				    mfg_id2 == mfg_id &&
				    version2 == version)
````

Reply:
> why ? we don't need following, just if (MASK_WITHOUT_UNIQUE_ID(adr) == MASK_WITHOUT_UNIQUE_ID(adr2))
> 
> ```
> 				mfg_id2 = SDW_MFG_ID(adr2);
> 				part_id2 = SDW_PART_ID(adr2);
> 				link_id2 = SDW_DISCO_LINK_ID(adr2);
> 				version2 = SDW_VERSION(adr2);
> 
> 				if (link_id2 == link_id &&
> 				    part_id2 == part_id &&
> 				    mfg_id2 == mfg_id &&
> 				    version2 == version)
> ```

Updated.